db/migration.go

```diff
@@ -107,8 +118,12 @@ func getMigrate(params params.Database) (*migrate.Migrate, error) {
 	defer func() {
 		_ = src.Close()
 	}()
-
-	m, err := migrate.NewWithSourceInstance("httpfs", src, params.ConnectionString)
+	var m *migrate.Migrate
+	if params.ConnectionString == "" {
+		m, err = getMigrateUsingEnvVariables(src)
```

Comment:
(Trying to learn migrate...) This ends up calling `NewWithInstance`, the other branch calls `NewWithSourceInstance`.  Can't we do the same in both and have a smaller `getMigrateUsingEnvVariables`?